src/core-asyncpipeline/AsyncPipeline/AsyncPipeline.T/Pipe/Pipe.Task.cs

```diff
@@ -0,0 +1,27 @@
+#nullable enable
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System
+{
+    partial struct AsyncPipeline<T>
+    {
+        public AsyncPipeline<TResult> Pipe<TResult>(Func<T, CancellationToken, Task<TResult>> mapAsync)
+            =>
+            InternalPipe(
+                mapAsync ?? throw new ArgumentNullException(nameof(mapAsync)));
+
+        internal AsyncPipeline<TResult> InternalPipe<TResult>(Func<T, CancellationToken, Task<TResult>> mapAsync)
+            =>
+            isCanceled || cancellationToken.IsCancellationRequested
+                ? new(valueTask: default, isCanceled: true, cancellationToken: cancellationToken)
+                : new(valueTask: InnerPipeAsync(mapAsync), isCanceled: isCanceled, cancellationToken: cancellationToken);
```

Comment:
Let's change `isCanceled: isCanceled` to `isCanceled: false` in the second strategy for readability reasons and to accord with the first (`isCanceled: true`) strategy.

(To look as strategies for canceled and not canceled cases more clear)

Reply:
Done